### DISCUSSION TITLE
generated taproot addresses

### DISCUSSION THREAD

**Discussion by punk3700** [ordinals/ord#1753] [General]
we're working on a wallet solution for our users who are less technical and are not familiar with the command line.  

we're going to generate a taproot address for each user.

1) is ord taproot address just like a regular btc taproot address or is there something more/different?

2) can each user's taproot address receive multiple inscriptions?

3) can the user send and receive inscriptions just from a regular taproot address?  

thanks!

**Comment by rot13maxi** [comment]
1. An `ord` taproot address is a plain old singlesig taproot address. You can receive an inscription to any bitcoin address (legacy, segwit, taproot). The tricky bits come when its time to send them. you either want to have coin control and spend just that UTXO, or have an ordinal-aware wallet.

2. an address can receive multiple inscriptions. Just make sure you send them in different transactions. If you send multiple inscriptions to the same address in the same tx, they'll end up in the same UTXO and will have to be split later (there isn't good tooling for this yet)

3. yes

**Reply by raphjaph** [reply]
3. yes, but if you wallet implements sending you have to watch out that the inscription isn't consumed by the fee. Our logic for sending  can found here: https://github.com/casey/ord/blob/master/src/subcommand/wallet/transaction_builder.rs
